vllm/utils.py

```diff
@@ -1550,6 +1550,62 @@ def import_from_path(module_name: str, file_path: Union[str, os.PathLike]):
     return module
 
 
+# Should handle every optional module in vLLM
+# This is based on `extras_require` from setup.py
+_EXTRAS_REQUIRE = {
+    "tensorizer": ["tensorizer"],
+    "runai": ["runai-model-streamer", "runai-model-streamer-s3", "boto3"],
+    "audio": ["librosa", "soundfile"],  # Required for audio processing
+    "video": ["decord"]  # Required for video processing
+}
```

Comment:
I think we can generate this through `pkg_resources` automatically, otherwise we need to update it manually when adding new extra requirements:
```python3
import pkg_resources

distribution = pkg_resources.get_distribution('vllm')
common_requirement = set(distribution.requires())
extras_requires = distribution.extras

_EXTRAS_REQUIRE = {}
for extra in extras_requires:
    requirements = set(distribution.requires([extra]))
    extra_requirements = requirements - common_requirement
    _EXTRAS_REQUIRE[extra] = [r.name for r in extra_requirements]

print(_EXTRAS_REQUIRE)

```

Reply:
Oh, just realized that python3.12 has deprecated `pkg_resources`, let me think a better method to handle this...

Reply:
Let me try this.

Reply:
Done